### PR TITLE
Docs for new Negotiate auth handler

### LINKED ISSUE #12420: Docs for new Negotiate auth handler
Docs for https://github.com/aspnet/AspNetCore/pull/9831

This may be an addition to or spinoff from https://docs.microsoft.com/en-us/aspnet/core/security/authentication/windowsauth?view=aspnetcore-2.2&tabs=visual-studio

This is a new authentication handler for Kerberos and NTLM and is intended for cross platform use.

WARNING: This component can persist credentials across requests on a given connection and MUST NOT be used with proxies unless they maintain a 1:1 connection affinity. This means it must not be used with Kestrel behind IIS (ANCM) out-of-proc.

Configuration:
- The code and options are quite minimal, but configuring your environment can be very complicated, especially for Kerbers, Linux, etc.. That's out of scope for this doc but we should link to some guides on environment setup.
- This only works with Kestrel.
- It is supported on Windows, Mac, and Linux
- We expect the primary usage to be Windows clients connecting to Linux servers that are part of the same Windows domain.
- HTTP/2 is not supported, but most clients automatically downgrade to HTTP/1.1 when challenged.

Results:
- On Windows this produces a WindowsPrincipal and WindowsIdentity.
- On Unix this produces a ClaimsIdentity with the name of the user (user@example.com). (No roles, etc.)

### PR DESCRIPTION
**Pull request by Tratcher**
Fixes #12420

[Doc preview](https://review.docs.microsoft.com/en-us/aspnet/core/security/authentication/windowsauth?view=aspnetcore-2.1&branch=pr-en-us-12793&tabs=visual-studio)

### CODE REVIEW

**Comment by Tratcher** [comment]
I know a few of the links don't work yet, they should start working as soon as preview6 gets published.

File: aspnetcore/security/authentication/windowsauth.md [hunk 1]
```diff
@@ -119,9 +132,66 @@ Use **either** of the following approaches:
...
+
+Some operating system specific configuration is required.
+
+#### Windows
+
+The [Microsoft.AspNetCore.Authentication.Negotiate](https://www.nuget.org/packages/Microsoft.AspNetCore.Authentication.Negotiate) component does user mode authentication so on Windows the SPNs must be added to the user account running the service, not the machine account. Call `setspn -S HTTP/mysrevername.mydomain.com myuser` from an admin command prompt.
+
+#### Linux and Mac
+
+Instructions for joining a Linux or Mac machine to a Windows domain can be found at [here](https://docs.microsoft.com/en-us/sql/azure-data-studio/enable-kerberos?view=sql-server-2017#join-your-os-to-the-active-directory-domain-controller). Those instructions create a machine account for the Linux machine on the domain. SPNs will need to be added to that SPN account.
```
Comment (Tratcher) [inline aspnetcore/security/authentication/windowsauth.md]:
```suggestion
Instructions for joining a Linux or Mac machine to a Windows domain can be found at [here](https://docs.microsoft.com/en-us/sql/azure-data-studio/enable-kerberos?view=sql-server-2017#join-your-os-to-the-active-directory-domain-controller). Those instructions create a machine account for the Linux machine on the domain. SPNs will need to be added to that account.
```

**Comment by guardrex** [comment]
I'll wait until the PU fully signs off. It's a faster if I apply the handful of grammar+style fixes directly.

@Tratcher I'll ping u after I do that  so that you can :eye: my commit and confirm final touches are ok.

**Comment by Tratcher** [comment]
@guardrex go ahead. Hao has already signed off and I added Barry more as an FYI.

**Comment by guardrex** [comment]
@Tratcher For sure, confirm that I didn't :boom: break any sentiment there. A few lines were a bit tricky to deal with.

File: aspnetcore/security/authentication/windowsauth.md [hunk 2]
```diff
@@ -119,9 +132,65 @@ Use **either** of the following approaches:
...
+
+Anonymous requests are allowed. Use [ASP.NET Core Authorization](xref:security/authorization/introduction) to challenge anonymous requests for authentication.
+
+### Windows environment configuration
+
+The [Microsoft.AspNetCore.Authentication.Negotiate](https://www.nuget.org/packages/Microsoft.AspNetCore.Authentication.Negotiate) component performs User Mode authentication. Service Principal Names (SPNs) must be added to the user account running the service, not the machine account. Execute `setspn -S HTTP/mysrevername.mydomain.com myuser` in an administrative command shell.
+
+### Linux and macOS environment configuration
+
+Instructions for joining a Linux or macOS machine to a Windows domain are available in the [Connect Azure Data Studio to your SQL Server using Windows authentication - Kerberos](/sql/azure-data-studio/enable-kerberos?view=sql-server-2017#join-your-os-to-the-active-directory-domain-controller) article. The instructions create a machine account for the Linux machine on the domain. SPNs must be added to the SPN account.
```
Comment (Tratcher) [inline aspnetcore/security/authentication/windowsauth.md]:
```suggestion
Instructions for joining a Linux or macOS machine to a Windows domain are available in the [Connect Azure Data Studio to your SQL Server using Windows authentication - Kerberos](/sql/azure-data-studio/enable-kerberos?view=sql-server-2017#join-your-os-to-the-active-directory-domain-controller) article. The instructions create a machine account for the Linux machine on the domain. SPNs must be added to that machine account.
```

File: aspnetcore/security/authentication/windowsauth.md [hunk 3]
```diff
@@ -119,9 +132,65 @@ Use **either** of the following approaches:
...
+
+> [!NOTE]
+> The Linux instructions work with Ubuntu 18.04. The install list is outdated for later versions of Ubuntu. Replace `python-software-properties` with `python3-software-properties` if needed.
```
Comment (Tratcher) [inline aspnetcore/security/authentication/windowsauth.md]: No, they were out of date even for 18.04.

File: aspnetcore/security/authentication/windowsauth.md [hunk 4]
```diff
@@ -169,7 +169,7 @@ The [Microsoft.AspNetCore.Authentication.Negotiate](https://www.nuget.org/packag
 Instructions for joining a Linux or macOS machine to a Windows domain are available in the [Connect Azure Data Studio to your SQL Server using Windows authentication - Kerberos](/sql/azure-data-studio/enable-kerberos?view=sql-server-2017#join-your-os-to-the-active-directory-domain-controller) article. The instructions create a machine account for the Linux machine on the domain. SPNs must be added to that machine account.
 
 > [!NOTE]
-> The Linux instructions were provided for Ubuntu 18.04, and the install list provided in the article is outdated. Replace `python-software-properties` with `python3-software-properties` if needed.
+> In the [Connect Azure Data Studio to your SQL Server using Windows authentication - Kerberos](/sql/azure-data-studio/enable-kerberos?view=sql-server-2017#join-your-os-to-the-active-directory-domain-controller) article, the Linux instructions were provided for Ubuntu 18.04. The install list provided in the article is outdated. Replace `python-software-properties` with `python3-software-properties` if needed.
```
Comment (Tratcher) [inline aspnetcore/security/authentication/windowsauth.md]: So confused... Those instructions don't specify an Ubuntu version. I tested them with 18.04 and they mostly worked except for this package name.

Comment (guardrex) [reply aspnetcore/security/authentication/windowsauth.md]: We could generalize it to say ...

> When following the guidance in the \[Connect Azure Data Studio to your SQL Server using Windows authentication - Kerberos](/sql/azure-data-studio/enable-kerberos?view=sql-server-2017#join-your-os-to-the-active-directory-domain-controller) article, replace `python-software-properties` with `python3-software-properties` if needed.

Comment (Tratcher) [reply aspnetcore/security/authentication/windowsauth.md]: Fair enough.

**Comment by guardrex** [comment]
I'll take a final look and merge when :rocket: *Commander Roth* 🚀 pulls the trigger.

**Comment by Tratcher** [comment]
> I'll take a final look and merge when 🚀 _Commander Roth_ 🚀 pulls the trigger.

Why wait?

**Comment by guardrex** [comment]
The links are broken, aren't they ... until we go live on Pre6?

**Comment by Tratcher** [comment]
The nuget links, yes, but they'll fix themselves.

**Comment by scottaddie** [comment]
@guardrex It's unlikely that we'll see a fix for that tab content validation error. Tabs are eventually being replaced with zone pivots.

**Comment by guardrex** [comment]
If the NuGet links don't work, then trying to use the package(s) will also fail, right?

I'm *too scared* 😨 to merge it in this state ... I'm *just a contractor!* :smile: lol

I'll let management do it ..... or you can.

**Comment by tdykstra** [comment]
> links don't work yet, they should start working as soon as preview6 gets published.

When will that happen?  We could go ahead and merge this, and just hold off on merging to Live.

**Comment by guardrex** [comment]
> when will that happen

It's variable ... 24 to a few hours prior to formal release.

> just hold off on merging to Live.

Yes, but one never knows when another team member is going to pull the trigger. I wouldn't risk it given my position here. I prefer management make the call on this.

File: aspnetcore/security/authentication/windowsauth.md [hunk 5]
```diff
@@ -119,9 +132,65 @@ Use **either** of the following approaches:
...
+
+### Linux and macOS environment configuration
+
+Instructions for joining a Linux or macOS machine to a Windows domain are available in the [Connect Azure Data Studio to your SQL Server using Windows authentication - Kerberos](/sql/azure-data-studio/enable-kerberos?view=sql-server-2017#join-your-os-to-the-active-directory-domain-controller) article. The instructions create a machine account for the Linux machine on the domain. SPNs must be added to that machine account.
+
+> [!NOTE]
+> When following the guidance in the [Connect Azure Data Studio to your SQL Server using Windows authentication - Kerberos](/sql/azure-data-studio/enable-kerberos?view=sql-server-2017#join-your-os-to-the-active-directory-domain-controller) article, replace `python-software-properties` with `python3-software-properties` if needed.
+
+Once the Linux or macOS machine is joined to the domain, additional steps are required to provide a keytab file with the SPNs:
```
Comment (blowdart) [inline aspnetcore/security/authentication/windowsauth.md]: Link keytab to https://blogs.technet.microsoft.com/pie/2018/01/03/all-you-need-to-know-about-keytab-files/?

File: aspnetcore/security/authentication/windowsauth.md [hunk 6]
```diff
@@ -119,9 +132,65 @@ Use **either** of the following approaches:
...
+
+* On the domain controller, add new web service SPNs to the machine account:
+  * `setspn -S HTTP/mywebservice.mydomain.com mymachine`
+  * `setspn -S HTTP/mywebservice@MYDOMAIN.COM mymachine`
+* Use [ktpass](/windows-server/administration/windows-commands/ktpass) to generate a keytab file:
+  * `ktpass -princ HTTP/mywebservice.mydomain.com@MYDOMAIN.COM -pass myKeyTabFilePassword -mapuser MYDOMAIN\mymachine$ -pType KRB5_NT_PRINCIPAL -out c:\temp\mymachine.HTTP.keytab -crypto AES256-SHA1`
+  * Some fields must be specified in uppercase as indicated.
+* Copy the keytab file to the Linux machine.
```
Comment (blowdart) [inline aspnetcore/security/authentication/windowsauth.md]: or macOS?

**Comment by Tratcher** [comment]
How long does it take docs to go live?

**Comment by guardrex** [comment]
errrr...... 🤔 ... I'm trying to get all of the Blazor PRs happily into master (and the sample app checked) first. ....errrrr .... say one hour. Is that ok?

**Comment by Tratcher** [comment]
Sure, just checking.

**Comment by guardrex** [comment]
I'm getting close ... so far, so good. 🤞

**Comment by guardrex** [comment]
@Tratcher :rocket: https://github.com/aspnet/AspNetCore.Docs/pull/12849

**Pull request merged**